xblock/runtime.py

```diff
@@ -973,13 +987,14 @@ def render_asides(self, block, view_name, frag, context):
         has the given block's rendering.
         """
         aside_frag_fns = []
-        for aside in self.get_asides(block):
-            aside_view_fn = aside.aside_view_declaration(view_name)
-            if aside_view_fn is not None:
-                aside_frag_fns.append((aside, aside_view_fn))
-        if aside_frag_fns:
-            # layout (overideable by other runtimes)
-            return self.layout_asides(block, context, frag, view_name, aside_frag_fns)
+        if block.supports_asides:
```

Comment:
The issue was that asides assumes that it has access to a ``usage_id`` in the block it is rendering on behalf of, and that was deeply embedded into the logic such that it seemed simpler to just not try to call it. IMO it makes sense that XBlock asides only render for XBlocks. I'll see if there's a more Pythonic way of doing this, as was discussed with my other ``supports_save`` property.